projects/components/src/datagrid/datagrid.component.spec.ts

```diff
@@ -884,6 +884,88 @@ describe('DatagridComponent', () => {
                 expect(this.clrGridWidget.component.columns.first.customFilter).toEqual(true);
             });
         });
+
```

Comment:
Please add unit test for the `columnsChanged` event. I would expect some test showing the difference about an event being fired when you set columns but not fired when you add/remove columns through the special methods?